chainer/backends/intel64.py

```diff
@@ -103,12 +103,15 @@ def inputs_all_ready(inputs, supported_ndim=(2, 4)):
         bool: ``True`` if all conditions meet.
 
     """
+
+    def check_array_type(a):
```

Comment:
Currently this function is used only from this module.
Could you make it private by prefixing with underscore (`_`)?

Also, `check_*` is not appropriate for a function that returns a logical value.
How about something like `_is_array_types_valid`, `_is_array_types_supported`, etc.?